chat-android/src/main/java/com/ably/chat/Typing.kt

```diff
@@ -258,6 +259,17 @@ internal class DefaultTyping(
      * Spec: CHA-T4, CHA-T14
      */
     override suspend fun keystroke() {
+        // CHA-T4e
+        if (room.realtimeClient.connection.state != ConnectionState.connected) {
```

Comment:
This check appears to be a suitable replacement for `room.ensureAttached`. Could you add a new method to the `room` class called `ensureConnected` and move this code into it? Also, you can update both `keyStroke` and `stop` to use `room.ensureConnected` instead of `room.ensureAttached`.

Reply:
Since both `Typing` and `Reactions` use ephemeral messages, just `CONNECTED` check seems suitable.